validation/src/main/java/io/serverlessworkflow/validation/WorkflowValidatorImpl.java

```diff
@@ -61,298 +62,272 @@ public WorkflowValidator setSource(String source) {
   }
 
   @Override
-  public List<ValidationError> validate() {
+  public Collection<ValidationError> validate() {
     validationErrors.clear();
-    if (workflow == null) {
+    if (workflow == null && schemaValidationEnabled && source != null) {
       try {
-        if (schemaValidationEnabled && source != null) {
-          try {
-            if (!source.trim().startsWith("{")) {
-              // convert yaml to json to validate
-              ObjectMapper yamlReader = new ObjectMapper(new YAMLFactory());
-              Object obj = yamlReader.readValue(source, Object.class);
-
-              ObjectMapper jsonWriter = new ObjectMapper();
-
-              workflowSchema.validate(new JSONObject(jsonWriter.writeValueAsString(obj)));
-            } else {
-              workflowSchema.validate(new JSONObject(source));
-            }
-          } catch (ValidationException e) {
-            e.getCausingExceptions().stream()
-                .map(ValidationException::getMessage)
-                .forEach(
-                    m -> {
-                      if ((!m.equals("#/functions: expected type: JSONObject, found: JSONArray")
-                          && !m.equals("#/events: expected type: JSONObject, found: JSONArray")
-                          && !m.equals("#/start: expected type: JSONObject, found: String")
-                          && !m.equals("#/retries: expected type: JSONObject, found: JSONArray"))) {
-                        addValidationError(m, ValidationError.SCHEMA_VALIDATION);
-                      }
-                    });
-          }
-        }
-      } catch (Exception e) {
-        logger.error("Schema validation exception: " + e.getMessage());
+        JsonSchemaFactory.getInstance(VersionFlag.V7)
+            .getSchema(workflowSchema)
+            .validate(Utils.getNode(source))
+            .forEach(m -> addValidationError(m.getMessage(), ValidationError.SCHEMA_VALIDATION));
+      } catch (IOException e) {
+        logger.error("Unexpected error during validation", e);
       }
     }
 
     // if there are schema validation errors
     // there is no point of doing the workflow validation
     if (validationErrors.size() > 0) {
       return validationErrors;
-    } else {
-      if (workflow == null) {
-        workflow = Workflow.fromSource(source);
-      }
+    } else if (workflow == null) {
+      workflow = Workflow.fromSource(source);
+    }
 
-      List<FunctionDefinition> functions =
-          workflow.getFunctions() != null ? workflow.getFunctions().getFunctionDefs() : null;
+    List<FunctionDefinition> functions =
```

Comment:
Can you check on these formatting changes? Iirc we have auto-formatting built in.

Reply:
This was changed automatically by the maven formatter when this class was changed. I guess it was not changed in a while